### PR TITLE
Allow the pcap packet size in pfdhcplistener to be configurable

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Allow the pcap packet size in pfdhcplistener to be configurable
## Impacts

pfdhcplistener

Enhancements
++++++++++++
- Allow the pcap packet size in pfdhcplistener to be configurable
## Delete branch after merge

NO

### CODE REVIEW

File: sbin/pfdhcplistener [hunk 1]
```diff
@@ -169,7 +169,7 @@ sub dhcp_detector {
     # updating process name so we know what interface we are listening on
     # WARNING: the format is expected by watchdog in pf::services. Don't change lightly.
     $PROGRAM_NAME = basename($PROGRAM_NAME) . ": listening on $interface";
-    $pcap = Net::Pcap::pcap_open_live( $interface, 576, 1, 0, \$err );
+    $pcap = Net::Pcap::pcap_open_live( $interface, $Config{advanced}{pfdhcplistener_packet_size} || 576, 1, 0, \$err );
```
Comment (julsemaan) [inline sbin/pfdhcplistener]: No need for `||` since we already defined it in pf.conf.defaults

Comment (rouzier) [inline sbin/pfdhcplistener]: The was more of a check to make sure that someone did not put 0.
But you are right we should just take it from pf.conf.defaults.
I will add a check in checkup.pm to not allow a size less than 576

Comment (julsemaan) [inline sbin/pfdhcplistener]: Even better

**Comment by jrouzierinverse** [comment]
Please review

File: lib/pf/pfcmd/checkup.pm [hunk 2]
```diff
@@ -1296,6 +1297,18 @@ sub cert_has_expired {
     return time > $expiration;
 }
 
+=item advanced_config
+
+Check the advanced config section
+
+=cut
+
+sub advanced_config {
```
Comment (julsemaan) [inline lib/pf/pfcmd/checkup.pm]: wouldn't it be better to check that all numeric fields defining a minimum are respecting it.

Would be less of a case by case and make it cleaner

Comment (jrouzierinverse) [inline lib/pf/pfcmd/checkup.pm]: True

**Comment by jrouzierinverse** [comment]
Please re-review

**Pull request merged**